### PR TITLE
[MRG] benchmarks across different GLM implementations

### PR DESCRIPTION
**Pull request by pavanramkumar**
A new convenience class `BenchmarkGLM` for users to run benchmarks on their own system. 

NB: The dependencies are complex (`scikit-learn` and `statsmodels are ok but `Rpy2` is not trivial to support), so it will not work out of the box.

### CODE REVIEW

**Comment by pavanramkumar** [comment]
Ok the dependencies are pretty complex and are affecting travis and circle CI. I tried a few things but there is no obvious quick fix for solving the `R` and `statsmodels` dependencies. 

If it's possible to exclude `benchmark.py` from test coverage we should do it.

This should close #8

**Comment by jasmainak** [comment]
yes you can exclude a file from testing :) I think you need to edit one of the config files. Probably like this: https://github.com/mne-tools/mne-python/blob/master/setup.cfg#L22

phew, github has recovered from the ddos ... at least for now

File: README.md [hunk 1]
```diff
@@ -53,6 +53,27 @@ but in contrast to these, Pyglmnet is a pure python implementation.
...
+
+Here is table comparing `pyglmnet` against
+`scikit-learn`'s `linear_model`, `statsmodels`, and `R`.
+
+The numbers below are run time (in milliseconds) to fit a
+$1000$ samples x $100$ predictors sparse matrix
+(density $0.05$). This was done on a c. 2011 Macbook Pro,
+so your numbers may vary.
+
+| distr     | pyglmnet   | sklearn  | statsmodels | R      |
```
Comment (jasmainak) [inline README.md]: sklearn -> scikit-learn

**Comment by pavanramkumar** [comment]
closes #8

**Pull request merged**